src/textual/css/_style_properties.py

```diff
@@ -290,6 +290,39 @@ def __set__(self, obj: Styles, spacing: str | None) -> str | None:
         return spacing
 
 
+class LayoutProperty:
+    """Descriptor for getting and setting layout."""
+
+    def __set_name__(self, owner: Styles, name: str) -> None:
+        self._internal_name = f"_rule_{name}"
+
+    def __get__(self, obj: Styles, objtype: type[Styles] | None = None) -> Layout:
+        """
+        Args:
+            obj (Styles): The Styles object
+            objtype (type[Styles]): The Styles class
+        Returns:
+            The ``Layout`` object.
+        """
+        return getattr(obj, self._internal_name)
+
+    def __set__(self, obj: Styles, layout: LayoutName | Layout):
+        """
+        Args:
+            obj (Styles): The Styles object.
+            layout (LayoutName | Layout): The layout to use. You can supply a ``LayoutName``
+                (a string literal such as ``"dock"``) or a ``Layout`` object.
+        """
+        from ..layouts.factory import get_layout, Layout  # Prevents circular import
+
+        obj.refresh(True)
```

Comment:
Can you set make keyword args explicity, i.e. `layout=True`

Reply:
Done (also fixed all other occurrences in this file 😄)